pkg/controller/clusterclaim/clusterclaim_controller.go

```diff
@@ -306,10 +307,15 @@ func (r *ReconcileClusterClaim) cleanupResources(claim *hivev1.ClusterClaim, log
 	}
 
 	// Delete ClusterDeployment
-	if cd.DeletionTimestamp == nil {
+	_, toRemove := cd.Annotations[constants.ClusterClaimRemoveClusterAnnotation]
+	if cd.DeletionTimestamp == nil && !toRemove {
 		logger.Info("deleting clusterDeployment")
-		if err := r.Delete(context.Background(), cd); err != nil {
-			logger.WithError(err).Log(controllerutils.LogLevel(err), "error deleting ClusterDeployment")
+		if cd.Annotations == nil {
+			cd.Annotations = map[string]string{}
+		}
+		cd.Annotations[constants.ClusterClaimRemoveClusterAnnotation] = ""
```

Comment:
I don't think a true/false value is valid here for this annotation.. since the presence provides the necessary information. true/false only adds additional error handling and possible break when someone sets some weird value here.

i'll change it to match the existing code.

Reply:
I agree but I've been nacked for it in the past. :)